substrate/runtime-support/src/dispatch.rs

```diff
@@ -84,13 +84,19 @@ macro_rules! decl_module {
 		pub struct $mod_type:ident<$trait_instance:ident: $trait_name:ident>;
 		$($rest:tt)*
 	) => {
+		// Workaround for https://github.com/rust-lang/rust/issues/26925 . Remove when sorted.
+		#[derive(Clone, Copy, PartialEq, Eq)]
+		#[cfg_attr(feature = "std", derive(Debug, Serialize, Deserialize))]
```

Comment:
IMO these derives should be on the struct themselves as opposed to within the macro with spooky action at a distance

Reply:
(the `$(#[$attr:meta])*` rule allows for this)